tracing-attributes/src/expand.rs

```diff
@@ -241,10 +241,13 @@ fn gen_block<B: ToTokens>(
             let level_tokens = event_args.level(Level::Error);
             match event_args.mode {
                 FormatMode::Default | FormatMode::Display => Some(quote!(
-                    tracing::event!(target: #target, #level_tokens, error = %e)
+                    tracing::event!(target: #target, #level_tokens, error = %e, "{}", e)
                 )),
                 FormatMode::Debug => Some(quote!(
-                    tracing::event!(target: #target, #level_tokens, error = ?e)
+                    tracing::event!(target: #target, #level_tokens, error = ?e, "{:?}", e)
+                )),
+                FormatMode::Raw => Some(quote!(
+                    tracing::event!(target: #target, #level_tokens, error = e, "{}", e)
```

Comment:
some types, such as `anyhow::Error`, don't actually implement `Error` directly, [but have `AsRef<&dyn Error>` implementations](https://docs.rs/anyhow/latest/anyhow/struct.Error.html#impl-AsRef%3Cdyn+Error+%2B+Sync+%2B+Send%3E-for-Error). is there anything we can do to make this mode also work nicely for events that have to be `as_ref`'d, without breaking returns for values which *are* `Error` implementers?

Reply:
That's an interesting area to think about. Presumably this falls into the problem domain of mapping attribute values to the `record_error(...)` hook within the `Layer` implementation?

I'm expecting that in this specific scenario we're not going to see too much by way of business logic around value types, though if I've missed something then please shout. As for implementing the business logic for handling `AsRef<&dyn Error>` I'm happy to chase that down, though I would propose landing that in a separate PR to keep the concepts neatly segregated.